src/main/webapp/app/shared/routes/route-selector.component.ts

```diff
@@ -76,12 +95,38 @@ export class JhiRouteSelectorComponent implements OnInit, OnDestroy {
     }
 
     private downRoute(instance: Route) {
-        if (instance && this.routes) {
-            const index = this.routes.findIndex((r) => r.appName === instance.appName);
-            if (index !== -1) {
-                this.routes[index].status = 'DOWN';
-            }
+        if (instance) {
+            instance.status = 'DOWN';
+        }
+    }
+
+    /** Change active time only if exists, else 0 **/
+    setActiveRefreshTime(time: number) {
+        if (time && this.refreshTimes.findIndex((t) => t === time) !== -1) {
+            this.activeRefreshTime = time;
+        } else {
+            this.activeRefreshTime = this.refreshTimes[0];
+        }
+        this.stateStorageService.storeSelectedRefreshTime(time);
+        this.launchTimer(true);
+    }
+
+    /** Init the timer **/
+    subscribe() {
+        if (this.activeRefreshTime && this.activeRefreshTime > 0) {
+            this.interval = Observable.interval(this.activeRefreshTime * 1000);
+            this.refreshTimer = this.interval.subscribe(() => {
```

Comment:
I'm not sure to understand. Have I to unsubscribe the "refreshTimer" ? (it's already the case, in ngOnDestroy)
Or other ?